client/opentracing/src/test/java/org/hawkular/apm/client/opentracing/SamplingTest.java

```diff
@@ -0,0 +1,185 @@
+                .asChildOf(rootSpan)
+                .start();
+
+        descendantZeroSampling.setTag(Tags.SAMPLING_PRIORITY.getKey(), 0);
+        carrier.clear();
+        tracer.inject(descendantZeroSampling.context(), Format.Builtin.TEXT_MAP, new TextMapInjectAdapter(carrier));
+        Assert.assertEquals(ReportingLevel.None.name(), carrier.get(Constants.HAWKULAR_APM_LEVEL));
+
+        descendantZeroSampling.finish();
+        descendant.finish();
+        rootSpan.finish();
+
+        Span descendantDescendantZeroSampling = tracer.buildSpan("bar")
+                .addReference(References.FOLLOWS_FROM, descendantZeroSampling.context())
+                .start();
+
+        carrier.clear();
+        tracer.inject(descendantDescendantZeroSampling.context(), Format.Builtin.TEXT_MAP, new TextMapInjectAdapter(carrier));
+        Assert.assertEquals(ReportingLevel.None.name(), carrier.get(Constants.HAWKULAR_APM_LEVEL));
+
+        descendantDescendantZeroSampling.finish();
+        Assert.assertEquals(1, traceRecorder.getTraces().size());
+    }
+
+    @Test
+    public void testSamplingPriorityChangedToOne() {
+        APMTracerTest.TestTraceRecorder traceRecorder = new APMTracerTest.TestTraceRecorder();
+        Tracer tracer = new APMTracer(traceRecorder, Sampler.ALWAYS_SAMPLE);
+
+        Span rootSpan = tracer.buildSpan("foo")
+                .asChildOf(extractedTraceState(tracer, ReportingLevel.None))
+                .start();
+
+        Span descendant = tracer.buildSpan("bar")
+                .asChildOf(rootSpan)
+                .start();
+
+        Map<String, String> carrier = new HashMap<>();
+        tracer.inject(descendant.context(), Format.Builtin.TEXT_MAP, new TextMapInjectAdapter(carrier));
+        Assert.assertEquals(ReportingLevel.None.name(), carrier.get(Constants.HAWKULAR_APM_LEVEL));
+
+        Span descendantZeroSampling = tracer.buildSpan("bar")
+                .asChildOf(rootSpan)
+                .start();
+
+        descendantZeroSampling.setTag(Tags.SAMPLING_PRIORITY.getKey(), 1);
+        carrier.clear();
+        tracer.inject(descendantZeroSampling.context(), Format.Builtin.TEXT_MAP, new TextMapInjectAdapter(carrier));
+        Assert.assertEquals(ReportingLevel.All.name(), carrier.get(Constants.HAWKULAR_APM_LEVEL));
+
+        descendantZeroSampling.finish();
+        descendant.finish();
+        rootSpan.finish();
+
+        Span descendantDescendantZeroSampling = tracer.buildSpan("bar")
+                .addReference(References.FOLLOWS_FROM, descendantZeroSampling.context())
+                .start();
+
+        carrier.clear();
+        tracer.inject(descendantDescendantZeroSampling.context(), Format.Builtin.TEXT_MAP, new TextMapInjectAdapter(carrier));
+        Assert.assertEquals(ReportingLevel.All.name(), carrier.get(Constants.HAWKULAR_APM_LEVEL));
+
+        descendantDescendantZeroSampling.finish();
+        Assert.assertEquals(2, traceRecorder.getTraces().size());
```

Comment:
This should be 1 - the first fragment had its 'rootSpan' sampling set to 0, so it should not have been recorded.